bitbots_blackboard/src/bitbots_blackboard/capsules/pathfinding_capsule.py

```diff
@@ -79,3 +87,170 @@ def get_current_pose(self):
+        if path_to_ball_service_response is None:
+            return
+        elif path_to_ball_service_response.done() and not self.path_updated:
+            self.path_updated = True
+            self.current_path_update = rospy.Time.now()
+            self.path_to_ball = path_to_ball_service_response.result().plan
+            time_to_ball = self.calculate_time_to_ball()
+            # path valid
+            if time_to_ball != -1:
+                self._blackboard.team_data.own_time_to_ball = time_to_ball
+                self.path_update_time = rospy.Time.now()
+                rospy.logerr("new path to ball")
+            # timeout
+            elif rospy.Time.now() - self.path_update_time >\
+                    rospy.Duration(self._blackboard.config['time_to_ball_remember_time']):
+                rospy.logerr("no path to ball found but path is too old")
+                self._blackboard.team_data.own_time_to_ball = 9999.0
+            else:
+                rospy.logerr("no path to ball found but i member")
+
+    def calculate_time_to_ball(self):
+        # calculate length of path
+        if len(self.path_to_ball.poses) > 2:
+            path_length = 0
+            for i in range(len(self.path_to_ball.poses)-1):
+                start = self.path_to_ball.poses[i].pose
+                end = self.path_to_ball.poses[i+1].pose
+                dist = math.sqrt((start.position.x-end.position.x) ** 2 + (start.position.y-end.position.y) ** 2)
+                path_length += dist
+
+            start_point = self.path_to_ball.poses[0].pose.position
+            end_point = self.path_to_ball.poses[-1].pose.position
+            straightline_distance = math.sqrt((start_point.x-end_point.x) ** 2 + (start_point.y-end_point.y) ** 2)
+            # if the robot is close to the ball it does not turn to walk to it
+            if straightline_distance < rospy.get_param("move_base/BBPlanner/orient_to_goal_distance", 1):
+                start_theta = euler_from_quaternion(numpify(self.path_to_ball.poses[0].pose.orientation))[2]
+                goal_theta = euler_from_quaternion(numpify(self.path_to_ball.poses[-1].pose.orientation))[2]
+                start_goal_theta_diff = abs(start_theta-goal_theta)
+                start_goal_theta_cost = start_goal_theta_diff * self._blackboard.config['time_to_ball_start_to_goal_angle_weight']
+                total_cost = path_length + start_goal_theta_cost
+                #rospy.logerr(f"Close to ball: start_goal_diff: {start_goal_theta_diff} " +
+                #             f"weighted start_goal_diff: {start_goal_theta_cost}, " +
+                #             f"path_length: {path_length}, " +
+                #             f"total: {total_cost}")
+            else:
+                # calculate how much we need to turn to start walking along the path
+                start_theta = euler_from_quaternion(numpify(self.path_to_ball.poses[0].pose.orientation))[2]
```

Comment:
The path-planning orients itself so its heading the goal position for a straightline_distance > orient_to_goal_distance and if straightline_distance < orient_to_goal_distance it rotates from the heading towards the goal orientation. The best approximation would therefore be:

```python

```